httpx/dispatch/http11.py

```diff
@@ -166,6 +166,9 @@ def __init__(
                     )
                 except OSError:  # pragma: nocover
                     data = b""
+                if self.h11_state.their_state == h11.SEND_RESPONSE and data == b"":
```

Comment:
Are we sure this is the right state? I thought that after a response is done the server would be in `h11.DONE`. cc: @njsmith @pgjones

Reply:
Presumably a moot point given https://github.com/encode/httpx/issues/96#issuecomment-515018947

Reply:
FWIW it is in that state on all my local tests, which I believe is correct since we do feed the request event to the state machine according to the [server state machine diagram](https://h11.readthedocs.io/en/latest/_static/SERVER.svg).

Reply:
You're right, I was looking at the bottom of the tree for our state without remember that by this point we'd already issued a request. So we should be checking for connection alive-ness before we get to that state then?

Reply:
Yep, stay tuned 😄